src/webcomponents/commons/forms/data-form.js

```diff
@@ -1578,6 +1621,91 @@ export default class DataForm extends LitElement {
         }
     }
 
+    _isFieldAutocomplete(field) {
+        // example: phenotypes[].1.description
+        if (field?.includes("[].")) {
+            const match = field.match(DataForm.re);
+            return !!(match && typeof this.dataAutocomplete?.[match?.groups?.arrayFieldName]?.[match?.groups?.index]?.[match?.groups?.field] !== "undefined");
+        }
+    }
```

Comment:
Please add a `return false` after the `if` condition, as this function should always return a boolean.